client/lib/core/widgets/media_settings_widget.dart

```diff
@@ -0,0 +1,475 @@
+
+  bool isLoading = true;
+  bool isLoadingCameraChange = false;
+
+  @override
+  void initState() {
+    super.initState();
+    initAll();
+
+    if (!widget.shouldShowVideoPreview) {
+      _videoElement = html.VideoElement();
+      isLoading = false;
+      return;
+    }
+
+    _videoElement = html.VideoElement()
+      ..id = _viewType
+      ..autoplay = true
+      ..muted = true
+      // Stops preview from going fullscreen on iOS Safari
+      ..setAttribute(
+        'playsinline',
+        'true',
+      )
+      ..setAttribute('webkit-playsinline', 'true')
+      ..setAttribute('disablePictureInPicture', 'true')
+      ..style.width = '100%'
+      ..style.height = '100%'
+      ..style.objectFit = 'cover'
+      // Mirror preview to match Agora local preview.
+      ..style.transform = 'scaleX(-1)';
+
+    ui_web.platformViewRegistry.registerViewFactory(
+      _viewType,
+      (int viewId) => _videoElement,
+    );
+  }
+
+  Future<void> initAll() async {
+    await _mediaService.init();
+
+    initialAudioDeviceId = _mediaService.selectedAudioInputId;
+    selectedAudioDeviceId = _mediaService.selectedAudioInputId;
+    initialVideoDeviceId = _mediaService.selectedVideoInputId;
+
+    userAudioEnabled = widget.conferenceRoom.audioEnabled;
+    userVideoEnabled = widget.conferenceRoom.videoEnabled;
+
+    await updatePreview();
+    if (!mounted) return;
+    setState(() {
+      isLoading = false;
+    });
+  }
+
+  Future<void> updatePreview() async {
+    if (!widget.shouldShowVideoPreview) return;
+    try {
+      await _mediaService.getUserMedia();
+      _videoElement.srcObject = _mediaService.previewMediaStream;
+    } catch (e) {
+      _videoElement.srcObject = null;
+    }
+  }
+
+  @override
+  void dispose() async {
+    super.dispose();
+    if (!widget.shouldShowVideoPreview) return;
+    _mediaService.stopPreviewMediaStream();
+    // If user doesn't save, we need to reset the video preview device.
+    // getUserMedia is called in updatePreview.
+    await _mediaService.selectVideoDevice(
+      deviceId: initialVideoDeviceId ?? '',
+      shouldUpdatePreview: false,
+    );
+    await updatePreview();
+    _videoElement.srcObject = null;
+    _videoElement.remove();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return AlertDialog(
+      insetPadding: responsiveLayoutService.isMobile(context)
+          ? EdgeInsets.symmetric(horizontal: 20.0, vertical: 12.0)
+          : null,
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Text(
+            'Audio Input Device',
+            style: context.theme.textTheme.titleMedium,
+          ),
+          _mediaService.audioInputs.isEmpty
+              ? Padding(
+                  padding: const EdgeInsets.only(top: 16.0),
+                  child: Text(
+                    'No audio devices available. Please check permissions.',
```

Comment:
I won't comment on each of these, but all the strings here and elsewhere should be pulled from [our l10n definitions](https://github.com/berkmancenter/frankly/tree/staging/client/lib/l10n), yeah?